test/e2e/base/provider/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ARG AGENT_TAG=9a05ad4b423a380c8da711e2a2b856ee93bc9422-es6
+ARG SW_VERSION=8.6.0
+
+FROM maven:3.8.1-jdk-8 as builder
+
+ADD . /provider_code
+
+WORKDIR /provider_code
+RUN mvn clean package -B -Dsw.version=8.6.0
+
+FROM ghcr.io/apache/skywalking/agent:${AGENT_TAG}
+
+VOLUME /services
+
+COPY --from=builder /provider_code/target/e2e-service-provider.jar /services/
+
+ENV JAVA_OPTS=""
+
+RUN echo 'plugin.toolkit.log.grpc.reporter.server_host=${SW_GRPC_LOG_SERVER_HOST:satellite}' >> /skywalking/agent/config/agent.config
+RUN echo 'plugin.toolkit.log.grpc.reporter.server_port=${SW_GRPC_LOG_SERVER_PORT:11800}' >> /skywalking/agent/config/agent.config
+RUN echo 'collector.get_profile_task_interval=${SW_COLLECTOR_GET_PROFILE_TASK_INTERVAL:1}' >> /skywalking/agent/config/agent.config
+RUN echo 'collector.get_agent_dynamic_config_interval=${SW_GET_AGENT_DYNAMIC_CONFIG_INTERVAL:1}' >> /skywalking/agent/config/agent.config
```

Comment:
Can we add the configuration of service instance here with the same way above, and remove `JAVA_OPTS` in the command and the `docker-compose.yml`?

The reason I want to get rid of `JAVA_OPTS` is that the test somewhat serves as an example of how we use SkyWalking Java agent, and I don't want our potential users think that they have to change their commands to use SkyWalking agent Docker image

Reply:
And finally we will add all the configs into agent.conf, so the `RUN echo ...` command will be removed

Reply:
Sure, have been move configurations to the `Dokerfile`